Exercises/NoModules/Chapter 10/Soln10_01.cpp

```diff
@@ -0,0 +1,38 @@
+// Exercise 10-1.cpp
+// Create a function template for my_clamp(), 
+// a template that produces functions to clamp values to a given interval.
+// Different variations are possible, but we opted for the same as in the Standard Library:
+// namely one where both arguments are of the same type, and passed by reference-to-const.
+#include <iostream>
+#include <format>
+#include <string>
+
+template<typename T> 
+const T& my_clamp(const T& a, const T& b, const T& c);    // Function template prototype
+
+int main() 
+{
+  std::cout << "2.0 clamped to interval [1.5, 2.5] is " << my_clamp(2.0, 1.5, 2.5) << std::endl;
+  std::cout << "5.0 clamped to interval [1.5, 2.5] is " << my_clamp(5.0, 1.5, 2.5) << std::endl;
+
+  int big_int {17011983}, small_int {10}, negative_int {-123};
+  std::cout 
+    << std::format("{} clamped to the interval [{},{}] is {}", 
+          negative_int, small_int, big_int, my_clamp(negative_int, big_int, small_int)) 
```

Comment:
This outputs:
`-123 clamped to the interval [10,17011983] is 17011983`
which looks wrong to me

Reply:
Ah, you parameters to my_clamp are reversed.
Should be:
`my_clamp(negative_int, small_int, big_int)`